src/main/java/edu/byu/cs/analytics/CommitAnalyticsRouter.java

```diff
@@ -0,0 +1,57 @@
+package edu.byu.cs.analytics;
+
+import edu.byu.cs.util.DateTimeUtils;
+import edu.byu.cs.util.FileUtils;
+
+import java.io.File;
+import java.time.Instant;
+
+/**
+ * Handles different type of commit analytics requests
+ */
+public class CommitAnalyticsRouter {
+
+    private static final String cacheDir = "commit-cache";
+
+    /**
+     * Updates the commit analytics CSV file and returns it
+     *
+     * @return the CSV data
+     */
+    public static String update() {
+        long ts = Instant.now().getEpochSecond();
+        String data = CommitAnalytics.generateCSV();
+
+        FileUtils.removeDirectory(new File(cacheDir));
+        FileUtils.createDirectory(cacheDir);
+        FileUtils.writeStringToFile(data, new File(cacheDir + "/" + ts + ".csv"));
```

Comment:
Just checking, will anything catastrophic happen if the file can't be created? It might be a good idea to create a specific error logging line for this

Reply:
If the file can't be created, a RuntimeException is thrown and the client gets the message, so nothing catastrophic. I still added a line to log the error, however, that's a good idea.